src/csharp/Grpc.Core/Grpc.Core.csproj

```diff
@@ -36,19 +36,19 @@
   <ItemGroup>
     <EmbeddedResource Include="..\..\..\etc\roots.pem" />
     <Content Include="..\nativelibs\csharp_ext_macos_x64\libgrpc_csharp_ext.dylib">
-      <PackagePath>runtimes/osx/native/libgrpc_csharp_ext.x64.dylib</PackagePath>
+      <PackagePath>runtimes/osx-x64/native/libgrpc_csharp_ext.x64.dylib</PackagePath>
```

Comment:
Changing this file is not enough, as there is other logic that relies on knowing these exact paths.

E.g. 
https://github.com/grpc/grpc/blob/9936d7785cdf41c13974497423c7d87bfbbd4054/src/csharp/Grpc.Core/build/net45/Grpc.Core.targets (which is required for running on legacy .NET frameworks other than Core CLR).

and
https://github.com/grpc/grpc/blob/9936d7785cdf41c13974497423c7d87bfbbd4054/src/csharp/Grpc.Core/Internal/NativeExtension.cs#L83

Reply:
Alright I have addressed these issues I think. For the core platforms (and xamarin too) the project should probably rely in the "smart" SDK for finding the correct native binary instead of creating the list itself. The runtime seems to be able handle the search paths (with all the fallback RIDs etc) and native library extensions (.so, .dll, dylib) by itself. I'm not sure how to access that logic with the way the native symbols are loaded currently, but it might be worth exploring.